src/pages/Filing/UpdateFinancialProfile/UfpForm.tsx

```diff
@@ -0,0 +1,156 @@
+import { scenarios } from 'pages/Summary/Summary.data';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Navigate, useParams } from 'react-router-dom';
+import getIsRoutingEnabled from 'utils/getIsRoutingEnabled';
+import type { InstitutionDetailsApiType } from '../ViewInstitutionProfile/institutionDetails.type';
+import AdditionalDetails from './AdditionalDetails';
+import FinancialInstitutionDetailsForm from './FinancialInstitutionDetailsForm';
+import UpdateAffiliateInformation from './UpdateAffiliateInformation';
+import UpdateIdentifyingInformation from './UpdateIdentifyingInformation';
+
+// TODO: defaultValues was causing the `value` provided to `<input>` fields to get wiped out.
+//   Figure out a smart way to combine them.
+// const defaultValues: UFPSchema = {
+//   tin: '',
+//   checkboxes: Object.fromEntries(
+//     checkboxOptions.map(option => [option.id, false]),
+//   ),
+// };
+export default function UFPForm({
+  data,
+}: {
+  data: InstitutionDetailsApiType;
+}): JSXElement {
+  const [submitted, setSubmitted] = useState(false);
+  const { lei } = useParams();
+  const isRoutingEnabled = getIsRoutingEnabled();
+
+  // TODO: Generate default values from `data`
+  const {
+    register,
+    control,
+    setValue,
+    trigger,
+    getValues,
+    formState: { errors: formErrors },
+  } = useForm<UFPSchema>({
+    resolver: zodResolver(ufpSchema),
+    // defaultValues,
+  });
+
+  // TODO: Render this based on the actual API call result
+  if (isRoutingEnabled && submitted) {
+    return (
+      <Navigate
+        replace
+        to='/summary'
+        state={{ scenario: scenarios.SuccessInstitutionProfileUpdate }}
+      />
+    );
+  }
+
+  // Used for error scrolling
+  const formErrorHeaderId = 'UFPFormErrorHeader';
+
+  // NOTE: This function is used for submitting the multipart/formData
+  const onSubmitButtonAction = async (): Promise<void> => {
+    const passesValidation = await trigger();
+    // TODO: Will be used for debugging after clicking 'Submit'
+    // eslint-disable-next-line no-console
+    console.log('passes validation?', passesValidation);
+    // if (passesValidation) {
+    const preFormattedData = getValues();
+    // TODO: Will be used for debugging after clicking 'Submit'
+    // eslint-disable-next-line no-console
+    console.log('data to be submitted (before format):', preFormattedData);
+
+    // POST formData
+    // TODO: Will be used for debugging after clicking 'Submit'
+    // eslint-disable-next-line no-console, @typescript-eslint/no-unused-vars
+    // const response = await submitUpdateFinancialProfile(
+    //   auth,
+    //   preFormattedData,
+    // )
+    // }
+    setSubmitted(true);
+  };
+
+  // TODO: Clear all checkboxes and inputs -- use setValue(defaultValues)
+  // eslint-disable-next-line no-console
+  const onClearform = (): void => console.log('onClearform clicked');
```

Comment:
Because the default values are coming from the API, I was struggling to get `react-hook-form` to correctly register the data once it was loaded.  I've adjusted the component structure and have this working (both defaultValues & reset()).